{'common': {'eval_train': False},
 'config': 'configs/exp1.yaml',
 'convert': False,
 'convert_nart': False,
 'count_op': False,
 'dataset': {'image_size': 224, 'uv_size': 256},
 'evaluate': False,
 'gpus': '0',
 'image_path': None,
 'model': {'dense_template_path': '../data/texture_data_256.npy',
           'face_eye_mask_path': '../data/uv_face_eye_mask.png',
           'face_mask_path': '../data/uv_face_mask.png',
           'fixed_displacement_path': '../data/fixed_displacement_256.npy',
           'flame_lmk_embedding_path': '../data/landmark_embedding.npy',
           'flame_model_path': '../data/generic_model.pkl',
           'jaw_type': 'aa',
           'max_z': 0.01,
           'mean_tex_path': '../data/mean_texture.jpg',
           'n_cam': 3,
           'n_detail': 128,
           'n_exp': 50,
           'n_light': 27,
           'n_pose': 6,
           'n_shape': 100,
           'n_tex': 50,
           'param_list': ['shape', 'tex', 'exp', 'pose', 'cam', 'light'],
           'tex_path': '../data/FLAME_albedo_from_BFM.npz',
           'tex_type': 'BFM',
           'topology_path': '../data/head_template.obj',
           'useTex': True,
           'use_tex': True,
           'uv_size': 256},
 'output_path': None,
 'pretrained_model': {'recog_network': 'net-data/Iter_007000_net.ckpt'},
 'recog_params': {'feature_dim': 512},
 'save_dict_path': 'snapshot/',
 'save_mesh_path': 'results/',
 'test': False,
 'test_params': {'batch_size': 3,
                 'crop_size': 224,
                 'iscrop': False,
                 'qualitative_test_list': 'utils/qualitative_test_images.txt',
                 'scale': 1.0,
                 'works': 1},
 'train_coarse': True,
 'train_detail': False,
 'train_params': {'batch_sizes': [64, 32],
                  'crop_size': 224,
                  'epoch_phase': 2,
                  'eye_closure_ldmk_idx_file': '../data/ldmk_eye_closure_idx.npy',
                  'eye_closure_loss_factors': [1.0, 1.0],
                  'iscrop': False,
                  'kpts_gt_dir': 'ldmk_gt',
                  'ldmk_loss_factors': [0.0001, 1.0],
                  'ldmk_weights_file': '../data/ldmk_weights.npy',
                  'norm_type_eye_closure': 'mse',
                  'norm_type_ldmk': 'mse',
                  'norm_type_photometric': 'mse',
                  'norm_type_reg': 'mse',
                  'optimizer': {'kwargs': {'lr': 0.0001,
                                           'weight_decay': 0.0005},
                                'type': 'Adam'},
                  'person_num_in_batch': 8,
                  'person_size': 250000,
                  'photometric_loss_factor': 2.0,
                  'regular_loss_factors': [0.0001, 0.0001],
                  'scale': 1.0,
                  'scheduler': {'epochs': 4,
                                'gamma': 0.5,
                                'milestones': [5, 10]},
                  'seg_masks_dir': 'face_seg_mask',
                  'shape_consistency_loss_factor': 1.0,
                  'size_per_person': 8,
                  'size_per_person_in_batch': 4,
                  'train_list': 'utils/train_images.txt',
                  'train_subject_list': 'utils/train_subject_images.txt',
                  'workers': 4},
 'visualize': False}
make save path: snapshot/configs/exp1.yaml_20210516_221025
make result path: results/configs/exp1.yaml_20210516_221025
fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
creating the FLAME Decoder
load recog model: net-data/Iter_007000_net.ckpt
total 2000000 images
train half iters: 15625.0
Build the code well for train_coarse
make save path: snapshot/configs/exp1.yaml_20210516_221049
make result path: results/configs/exp1.yaml_20210516_221049
fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
creating the FLAME Decoder
load recog model: net-data/Iter_007000_net.ckpt
total 2000000 images
train half iters: 15625.0
Build the code well for train_coarse
Training ..... 
main.py:39: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  config = yaml.load(f)
/mnt/lustre/wuxingzhe/anaconda3.5.1/envs/deca/lib/python3.8/site-packages/pytorch3d/io/obj_io.py:70: UserWarning: Faces have invalid indices
  warnings.warn("Faces have invalid indices")
/mnt/lustre/wuxingzhe/anaconda3.5.1/envs/deca/lib/python3.8/site-packages/pytorch3d/io/obj_io.py:484: UserWarning: Mtl file does not exist: ../data/template.mtl
  warnings.warn(f"Mtl file does not exist: {f_mtl}")
/mnt/lustre/wuxingzhe/anaconda3.5.1/envs/deca/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:117: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
Traceback (most recent call last):
  File "main.py", line 60, in <module>
    main()
  File "main.py", line 49, in main
    agent.trainval()
  File "/mnt/lustre/wuxingzhe/DECA_Pytorch/decalib/deca_solver.py", line 246, in trainval
    self.train(epoch)
  File "/mnt/lustre/wuxingzhe/DECA_Pytorch/decalib/deca_solver.py", line 182, in train
    output = self.decode(parameters, epoch)
  File "/mnt/lustre/wuxingzhe/DECA_Pytorch/decalib/deca_solver.py", line 107, in decode
    albedo = self.flametex(codedict['tex'])
  File "/mnt/lustre/wuxingzhe/anaconda3.5.1/envs/deca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "/mnt/lustre/wuxingzhe/DECA_Pytorch/decalib/models/FLAME.py", line 258, in forward
    texture = self.texture_mean + (self.texture_basis*texcode[:,None,:]).sum(-1)
RuntimeError: CUDA out of memory. Tried to allocate 9.38 GiB (GPU 0; 15.75 GiB total capacity; 5.60 GiB already allocated; 9.06 GiB free; 5.69 GiB reserved in total by PyTorch)
srun: error: SH-IDC1-10-5-37-43: task 0: Exited with exit code 1
